native-sql-engine/cpp/src/codegen/arrow_compute/ext/sort_kernel.cc

```diff
@@ -2133,23 +2829,36 @@ SortArraysToIndicesKernel::SortArraysToIndicesKernel(
 #undef PROCESS_SUPPORTED_TYPES
 
 arrow::Status SortArraysToIndicesKernel::Evaluate(ArrayList& in) {
-  return impl_->Evaluate(in);
+  const std::lock_guard<std::mutex> lock(spill_lck_);
+  RETURN_NOT_OK(impl_->Evaluate(in));
+  RETURN_NOT_OK(impl_->MaybeSpill());
```

Comment:
Do this 2 lines acquire for memory from context memory pool too?

Reply:
MaybeSpill will check if cached data go beyond threshold, and then do spill and read the first batch from disk, so there maybe some allocation happened.

Reply:
So this may cause deadlock? Somewhere seemed hanging when I was testing the code. I made sure it was not a Spill->ReadBatch->Spill call. maybe it was MaybeSpill->ReadBatch->Spill

Reply:
maybespill is spill